app/public/model/UserPreferences.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { Observable } from '/js/src/index.js';
+import { replaceUrlParams } from '../utils/url/urlUtils.js';
+import { RCT } from '../config.js';
+const { dataReqParams } = RCT;
+
+const defaultRowsOnSite = 50;
+
+export const sidebarPreferences = {
+    visible: 'visible',
+    collapsible: 'collapsible',
+};
+
+/** Rows per page, UI theme, Sidebar, Detector list, predefined filters */
+
+export default class UserPreferences extends Observable {
+    constructor(parent) {
+        super();
```

Comment:
if UserPreferences is supposed to keep global preferences for e.g. rows-on-site, then href's in `linkChips` in `pageCellSpecialls` should be changed e.g. app/public/views/userView/data/pagesCellsSpecials.js:54, because they shadow UserPreferences. If it is view specific, then number in setting dialog should be the same as number of rows on current page, see screenchot
![Screenshot 2023-08-10 at 17 05 41](https://github.com/AliceO2Group/RunConditionTable/assets/65893715/2b70e931-cc9a-4625-887d-ee65401ff7c9)

Reply:
Done, thanks for spotting that!